TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Command/NodeCommandControllerPlugin.php

```diff
@@ -1005,9 +1005,11 @@ protected function collectNodesWithInvalidDimensions($workspaceName, array $allo
             foreach ($allowedDimensionCombinations as $dimensionConfiguration) {
                 ksort($dimensionConfiguration);
                 ksort($nodeDataArray['dimensionValues']);
-                if ($dimensionConfiguration === $nodeDataArray['dimensionValues']) {
-                    $foundValidDimensionValues = true;
-                    break;
+                foreach ($dimensionConfiguration as $dimensionKey => $dimensionValuesArray) {
+                    if (in_array($nodeDataArray['dimensionValues'][$dimensionKey][0], $dimensionValuesArray)) {
```

Comment:
Well, if any of the keys `'dimensionValues'`, ' $dimensionKey` or `0` are not set, it will produce `Notice: Undefined index: … in …` Now it depends on what can happen at that place. `dimensionValues` must always exist, but from then on the keys could not be set, I'd say.